lib/en.js

```diff
@@ -7,9 +10,32 @@ const COMMON_NONASCII_PUNCTUATION = `…“”‘’`;
 const COMMON_PUNCTUATION = ASCII_PUNCTUATION + COMMON_NONASCII_PUNCTUATION;
 const COMMON_SYMBOLS = ASCII_SYMBOLS;
 
+const commonCombiningDiacritics = [
+  '\u0300', // grave accent
+  '\u0301', // acute accent
+  '\u0302', // circumflex
+  '\u0303', // tilde
+  '\u0308', // diaresis
+  '\u0327', // cedilla
+];
+
+const COMMON_DIACRITICS = new FunctionRule((s1, s2) => {
+  let s1prime = normalize(s1, 'NFD');
+  let s2prime = normalize(s2, 'NFD');
+
+  commonCombiningDiacritics.forEach(c => {
+    s1prime = replaceAll(s1prime, c);
+    s2prime = replaceAll(s2prime, c);
+  });
+
+  // FIXME: should be restored to whatever normalization they came in with.
+  return [normalize(s1prime, 'NFC'), normalize(s2prime, 'NFC')];
```

Comment:
Just to make sure I understand - it looks like this may break the match if we want unicode normalization to matter? If that's true, I wonder how we might work around that. Perhaps relying on some unicode normalization-agnostic ascii folding? Or maybe we actually want to bake "unicode normalization never matters" into this library if there are truly never any cases in which we want that to count?

Reply:
Yes, good eye. This issue was already present in the accents rule: https://github.com/spanishdict/equivalency/blob/master/lib/index.js#L120 

A few thoughts on approaches:
- I was thinking about making `doesntMatter(UNICODE_NORMALIZATION)` be the (only) default rule since it seems like a sane setting for most possible use cases. However, ideally I'd still like to be able to offer `matters(UNICODE_NORMALIZATION)`.
- When unicode normalization matters, one approach is to make that be the first thing that is checked, and short-circuit return `isEquivalent: false` is that's the case. But I'd like to eventually return supplementary information about what differences were found, so to support that, instead of short-circuit returning, a flag could be set when a normalization difference is found, and then the rest of the equivalency checks should be free to normalize as needed. 
- It should be possible to do something like keep track of the indexes of which codepoints were affected by removing accents or diacritics, then after that operation is done, restore all the original codepoints that weren't affected and return those strings -- preserving the original normalization, even if it were, say, a mix of NFC and NFD. But this seems more complex and error-prone and I'm not sure it adds value compared to the previous approach, so I'm inclined not to go this route until/unless there is a concrete need for it.

What do you think?

Whatever approach we go with, I'd like it to be tackled in a separate PR.